Initial docs for the csp adapter (#4143)
rancher/docs: content/rancher/v2.6/en/cloud-marketplace/supportconfig/_index.md

---
title: Supportconfig bundle 
weight: 100
---

After installing the CSP adapter, you will have the ability to generate a supportconfig bundle. This bundle is a tar file which can be used to quickly provide information to support.

> These bundles can be created through rancher or through direct access to the cluster rancher is installed on (accessing through rancher is preferred).

Author: Thank you for the suggestion, change made

> > **Note:** Only admin users can generate/download supportconfig bundles, regardless of method 

Author: Thank you for the suggestion, change made

> ## Accessing through Rancher

Reviewer: ```suggestion
### Accessing through Rancher
```

Author: Thank you for the suggestion, change made

First, click on the hamburger menu. Then click the `Get Support` button.

![Get Support]({{<baseurl>}}/img/rancher/support-help.png)

In the next page, click on the `Generate Support Config` button.

> > **Note:** If the adapter is not installed, the option to generate the support-config bundle will not be present. You must install the csp adapter to generate a support-config bundle.

Reviewer 2: ```suggestion
> **Note:** If the adapter is not installed, the option to generate the supportconfig bundle will not be present. You must install the CSP adapter to generate a supportconfig bundle.
```

Author: Thank you for the suggestion, change made

![Get Support]({{<baseurl>}}/img/rancher/generate-support-config.png)

> ## Accessing without rancher

Reviewer: ```suggestion
### Accessing without Rancher
```

Reviewer: Keeping the header sizes consistent.

Author: Thank you for the suggestion, change made

> First, generate a kubeconfig for the cluster rancher is installed on. 

Reviewer: ```suggestion
First, generate a kubeconfig for the cluster that Rancher is installed on. 
```

> > **Note:** If rancher is down, you will not be able to use a kubeconfig token generated by rancher to access the cluster.

Reviewer: ```suggestion
> **Note:** If Rancher is down, you will not be able to use a kubeconfig token generated by Rancher to access the cluster.
```

Author: Thank you for the suggestion, change made

Configure your shell environment to use this kubeconfig token:

```bash
export KUBECONFIG=$MY_KUBECONFIG_PATH
```

It is recommended to create a temporary working directory while running this command, like below:

```bash
mkdir temp && cd temp
```

> Then, retrieve the supportconfig bundle.

Reviewer: ```suggestion
Then, retrieve the supportconfig bundle:
```

Author: Thank you for the suggestion, change made

```bash
mkdir rancher && kubectl get configmap csp-config -n cattle-csp-adapter-system -o=jsonpath='{.data.data}' >> rancher/config.json && tar -c -f supportconfig_rancher.tar rancher && rm -rf rancher
```

This will create a `supportconfig_rancher.tar` file in your current directory.

> Users who run these commands on Mac may experience issues due incompatibilities between gnu-tar and bsd-tar. If support has issues reading a supportconfig that you produce, try using the below command after making gnu-tar accessible as `gtar` on your path:

Reviewer 2: ```suggestion
Users who run these commands on Mac may experience issues due to incompatibilities between gnu-tar and bsd-tar. If support has issues reading a supportconfig that you produce, try using the below command after making gnu-tar accessible as `gtar` on your path:
```

Author: Thank you for the suggestion, change made
